The failures are genuine and due to the fact that I mistakenly added a
change to Redis.
I'll update the PR soon

On Wed, Aug 25, 2021, 19:51 quarkus-bot[bot] ***@***.***>
wrote:

> Failing Jobs - Building 236900a
> <https://github.com/quarkusio/quarkus/commit/236900afc0d1affc07131d430fc5b4c059b1c3c8>
> Status Name Step Test failures Logs Raw logs
> ✖ JVM Tests - JDK 11 Build Test failures
> <#m_3094285745310865949_user-content-test-failures-job-3422489883> Logs
> <https://github.com/quarkusio/quarkus/runs/3422489883?check_suite_focus=true> Raw
> logs
> <https://github.com/quarkusio/quarkus/commit/236900afc0d1affc07131d430fc5b4c059b1c3c8/checks/3422489883/logs>
> ✖ JVM Tests - JDK 16 Build Test failures
> <#m_3094285745310865949_user-content-test-failures-job-3422489948> Logs
> <https://github.com/quarkusio/quarkus/runs/3422489948?check_suite_focus=true> Raw
> logs
> <https://github.com/quarkusio/quarkus/commit/236900afc0d1affc07131d430fc5b4c059b1c3c8/checks/3422489948/logs>
> ✖ MicroProfile TCKs Tests Verify Test failures
> <#m_3094285745310865949_user-content-test-failures-job-3422490293> Logs
> <https://github.com/quarkusio/quarkus/runs/3422490293?check_suite_focus=true> Raw
> logs
> <https://github.com/quarkusio/quarkus/commit/236900afc0d1affc07131d430fc5b4c059b1c3c8/checks/3422490293/logs>
>
> Full information is available in the Build summary check run
> <https://github.com/quarkusio/quarkus/runs/3424467244>.
> Test Failures ⚙️ JVM Tests - JDK 11 #
> <#m_3094285745310865949_user-content-test-failures-job-3422489883> 📦
> integration-tests/container-image/maven-invoker-way/target/it/container-image-jib-with-redis
>
> ✖ org.acme.redis.IncrementResourceIT.testRedisOperations - More details
> <https://github.com/quarkusio/quarkus/runs/3424467244#user-content-test-failure-org.acme.redis.incrementresourceit-1>
> - Source on GitHub
> <https://github.com/geoand/quarkus/blob/236900afc0d1affc07131d430fc5b4c059b1c3c8/integration-tests/container-image/maven-invoker-way/target/it/container-image-jib-with-redis/src/test/java/org/acme/redis/IncrementResourceIT.java>
>
> java.lang.RuntimeException:
>
> java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
>
> 	[error]: Build step io.quarkus.redis.client.deployment.DevServicesRedisProcessor#startRedisContainers threw an exception: org.testcontainers.containers.ContainerLaunchException: Container startup failed
>
> 	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:334)
>
> 	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:315)
>
> 	at io.quarkus.redis.client.deployment.DevServicesRedisProcessor.lambda$startContainer$1(DevServicesRedisProcessor.java:160)
>
> 	at java.base/java.util.Optional.orElseGet(Optional.java:369)
>
> 	at io.quarkus.redis.client.deployment.DevServicesRedisProcessor.startContainer(DevServicesRedisProcessor.java:168)
>
> 	at io.quarkus.redis.client.deployment.DevServicesRedisProcessor.startRedisContainers(DevServicesRedisProcessor.java:90)
>
> 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invok...
>
>
> ------------------------------
> ⚙️ JVM Tests - JDK 16 #
> <#m_3094285745310865949_user-content-test-failures-job-3422489948> 📦
> integration-tests/container-image/maven-invoker-way/target/it/container-image-jib-with-redis
>
> ✖ org.acme.redis.IncrementResourceIT.testRedisOperations - More details
> <https://github.com/quarkusio/quarkus/runs/3424467244#user-content-test-failure-org.acme.redis.incrementresourceit-1>
> - Source on GitHub
> <https://github.com/geoand/quarkus/blob/236900afc0d1affc07131d430fc5b4c059b1c3c8/integration-tests/container-image/maven-invoker-way/target/it/container-image-jib-with-redis/src/test/java/org/acme/redis/IncrementResourceIT.java>
>
> java.lang.RuntimeException:
>
> java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
>
> 	[error]: Build step io.quarkus.redis.client.deployment.DevServicesRedisProcessor#startRedisContainers threw an exception: org.testcontainers.containers.ContainerLaunchException: Container startup failed
>
> 	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:334)
>
> 	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:315)
>
> 	at io.quarkus.redis.client.deployment.DevServicesRedisProcessor.lambda$startContainer$1(DevServicesRedisProcessor.java:160)
>
> 	at java.base/java.util.Optional.orElseGet(Optional.java:364)
>
> 	at io.quarkus.redis.client.deployment.DevServicesRedisProcessor.startContainer(DevServicesRedisProcessor.java:168)
>
> 	at io.quarkus.redis.client.deployment.DevServicesRedisProcessor.startRedisContainers(DevServicesRedisProcessor.java:90)
>
> 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invok...
>
>
> ------------------------------
> ⚙️ MicroProfile TCKs Tests #
> <#m_3094285745310865949_user-content-test-failures-job-3422490293> 📦
> tcks/microprofile-fault-tolerance
>
> ✖
> org.eclipse.microprofile.fault.tolerance.tck.TimeoutUninterruptableTest.testTimeoutAsyncBulkhead
> line 190 - More details
> <https://github.com/quarkusio/quarkus/runs/3424467244#user-content-test-failure-org.eclipse.microprofile.fault.tolerance.tck.timeoutuninterruptabletest-1>
> - Source on GitHub
> <https://github.com/geoand/quarkus/blob/236900afc0d1affc07131d430fc5b4c059b1c3c8/tcks/microprofile-fault-tolerance/src/test/java/org/eclipse/microprofile/fault/tolerance/tck/TimeoutUninterruptableTest.java#L190>
>
> java.lang.AssertionError: Unexpected exception thrown from Future
>
> 	at org.testng.Assert.fail(Assert.java:85)
>
> 	at org.eclipse.microprofile.fault.tolerance.tck.util.Exceptions.expect(Exceptions.java:98)
>
> 	at org.eclipse.microprofile.fault.tolerance.tck.TimeoutUninterruptableTest.testTimeoutAsyncBulkhead(TimeoutUninterruptableTest.java:190)
>
> 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>
> 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
>
> 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
>
> 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
>
> 	at io.quarkus.arquillian.QuarkusProtocol$QuarkusMethodExecutor$1.invoke(QuarkusProtocol.java:87)
>
> 	at org.jboss.arquillian.container.test.impl.execution.LocalTestExecuter.execute(LocalTestExecuter.java:57)
>
> 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>
> 	at java.base...
>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/quarkusio/quarkus/pull/19648#issuecomment-905705885>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABBMDPY6AJOPLIJDMPRQX23T6UNRFANCNFSM5CYV6NWA>
> .
>

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -203,8 +203,8 @@
         <avro.version>1.10.2</avro.version>
         <apicurio-registry.version>2.0.1.Final</apicurio-registry.version>
         <jacoco.version>0.8.7</jacoco.version>
-        <testcontainers.version>1.15.3</testcontainers.version>
-        <docker-java.version>3.2.8</docker-java.version> <!-- must be the version Testcontainers use -->
+        <testcontainers.version>1.16.0</testcontainers.version>
+        <docker-java.version>3.2.11</docker-java.version> <!-- must be the version Testcontainers use -->
         <aesh-readline.version>2.1</aesh-readline.version>
     </properties>
 

--- a/extensions/devservices/common/src/main/java/io/quarkus/devservices/common/ConfigureUtil.java
+++ b/extensions/devservices/common/src/main/java/io/quarkus/devservices/common/ConfigureUtil.java
@@ -19,9 +19,6 @@ public final class ConfigureUtil {
         String hostName = hostNamePrefix + "-" + Base58.randomString(5);
         container.setNetworkAliases(Collections.singletonList(hostName));
 
-        // we need to clear the exposed ports as they don't make sense when the application is going to
-        // to be communicating with the DB over the same network
-        container.setExposedPorts(Collections.emptyList());
         return hostName;
     }
 }

--- a/extensions/devservices/db2/src/main/java/io/quarkus/devservices/db2/deployment/DB2DevServicesProcessor.java
+++ b/extensions/devservices/db2/src/main/java/io/quarkus/devservices/db2/deployment/DB2DevServicesProcessor.java
@@ -35,17 +35,17 @@ public class DB2DevServicesProcessor {
             public RunningDevServicesDatasource startDatabase(Optional<String> username, Optional<String> password,
                     Optional<String> datasourceName, Optional<String> imageName, Map<String, String> additionalProperties,
                     OptionalInt fixedExposedPort, LaunchMode launchMode) {
-                Db2Container container = new QuarkusDb2Container(imageName, fixedExposedPort,
-                        devServicesSharedNetworkBuildItem.isPresent())
-                                .withPassword(password.orElse("quarkus"))
-                                .withUsername(username.orElse("quarkus"))
-                                .withDatabaseName(datasourceName.orElse("default"));
+                QuarkusDb2Container container = new QuarkusDb2Container(imageName, fixedExposedPort,
+                        devServicesSharedNetworkBuildItem.isPresent());
+                container.withPassword(password.orElse("quarkus"))
+                        .withUsername(username.orElse("quarkus"))
+                        .withDatabaseName(datasourceName.orElse("default"));
                 additionalProperties.forEach(container::withUrlParam);
                 container.start();
 
                 LOG.info("Dev Services for IBM Db2 started.");
 
-                return new RunningDevServicesDatasource(container.getJdbcUrl(), container.getUsername(),
+                return new RunningDevServicesDatasource(container.getEffectiveJdbcUrl(), container.getUsername(),
                         container.getPassword(),
                         new Closeable() {
                             @Override
@@ -86,8 +86,11 @@ public class DB2DevServicesProcessor {
             }
         }
 
-        @Override
-        public String getJdbcUrl() {
+        // this is meant to be called by Quarkus code and is not strictly needed
+        // in the DB2 case as testcontainers does not try to establish
+        // a connection to determine if the container is ready, but we do it anyway to be consistent across
+        // DB containers
+        public String getEffectiveJdbcUrl() {
             if (useSharedNetwork) {
                 // in this case we expose the URL using the network alias we created in 'configure'
                 // and the container port since the application communicating with this container

--- a/extensions/devservices/mariadb/src/main/java/io/quarkus/devservices/mariadb/deployment/MariaDBDevServicesProcessor.java
+++ b/extensions/devservices/mariadb/src/main/java/io/quarkus/devservices/mariadb/deployment/MariaDBDevServicesProcessor.java
@@ -33,17 +33,17 @@ public class MariaDBDevServicesProcessor {
             public RunningDevServicesDatasource startDatabase(Optional<String> username, Optional<String> password,
                     Optional<String> datasourceName, Optional<String> imageName, Map<String, String> additionalProperties,
                     OptionalInt fixedExposedPort, LaunchMode launchMode) {
-                MariaDBContainer container = new QuarkusMariaDBContainer(imageName, fixedExposedPort,
-                        devServicesSharedNetworkBuildItem.isPresent())
-                                .withPassword(password.orElse("quarkus"))
-                                .withUsername(username.orElse("quarkus"))
-                                .withDatabaseName(datasourceName.orElse("default"));
+                QuarkusMariaDBContainer container = new QuarkusMariaDBContainer(imageName, fixedExposedPort,
+                        devServicesSharedNetworkBuildItem.isPresent());
+                container.withPassword(password.orElse("quarkus"))
+                        .withUsername(username.orElse("quarkus"))
+                        .withDatabaseName(datasourceName.orElse("default"));
                 additionalProperties.forEach(container::withUrlParam);
                 container.start();
 
                 LOG.info("Dev Services for MariaDB started.");
 
-                return new RunningDevServicesDatasource(container.getJdbcUrl(), container.getUsername(),
+                return new RunningDevServicesDatasource(container.getEffectiveJdbcUrl(), container.getUsername(),
                         container.getPassword(),
                         new Closeable() {
                             @Override
@@ -84,8 +84,9 @@ public class MariaDBDevServicesProcessor {
             }
         }
 
-        @Override
-        public String getJdbcUrl() {
+        // this is meant to be called by Quarkus code and is needed in order to not disrupt testcontainers
+        // from being able to determine the status of the container (which it does by trying to acquire a connection)
+        public String getEffectiveJdbcUrl() {
             if (useSharedNetwork) {
                 String additionalUrlParams = constructUrlParameters("?", "&");
                 return "jdbc:mariadb://" + hostName + ":" + PORT + "/" + getDatabaseName() + additionalUrlParams;

--- a/extensions/devservices/mssql/src/main/java/io/quarkus/devservices/mssql/deployment/MSSQLDevServicesProcessor.java
+++ b/extensions/devservices/mssql/src/main/java/io/quarkus/devservices/mssql/deployment/MSSQLDevServicesProcessor.java
@@ -7,7 +7,6 @@ import java.util.Optional;
 import java.util.OptionalInt;
 
 import org.jboss.logging.Logger;
-import org.testcontainers.containers.JdbcDatabaseContainer;
 import org.testcontainers.containers.MSSQLServerContainer;
 import org.testcontainers.utility.DockerImageName;
 
@@ -36,15 +35,15 @@ public class MSSQLDevServicesProcessor {
             public RunningDevServicesDatasource startDatabase(Optional<String> username, Optional<String> password,
                     Optional<String> datasourceName, Optional<String> imageName, Map<String, String> additionalProperties,
                     OptionalInt fixedExposedPort, LaunchMode launchMode) {
-                JdbcDatabaseContainer container = new QuarkusMSSQLServerContainer(imageName, fixedExposedPort,
-                        devServicesSharedNetworkBuildItem.isPresent())
-                                .withPassword(password.orElse("Quarkuspassword1"));
+                QuarkusMSSQLServerContainer container = new QuarkusMSSQLServerContainer(imageName, fixedExposedPort,
+                        devServicesSharedNetworkBuildItem.isPresent());
+                container.withPassword(password.orElse("Quarkuspassword1"));
                 additionalProperties.forEach(container::withUrlParam);
                 container.start();
 
                 LOG.info("Dev Services for Microsoft SQL Server started.");
 
-                return new RunningDevServicesDatasource(container.getJdbcUrl(), container.getUsername(),
+                return new RunningDevServicesDatasource(container.getEffectiveJdbcUrl(), container.getUsername(),
                         container.getPassword(),
                         new Closeable() {
                             @Override
@@ -86,8 +85,9 @@ public class MSSQLDevServicesProcessor {
             }
         }
 
-        @Override
-        public String getJdbcUrl() {
+        // this is meant to be called by Quarkus code and is needed in order to not disrupt testcontainers
+        // from being able to determine the status of the container (which it does by trying to acquire a connection)
+        public String getEffectiveJdbcUrl() {
             if (useSharedNetwork) {
                 // in this case we expose the URL using the network alias we created in 'configure'
                 // and the container port since the application communicating with this container

--- a/extensions/devservices/mysql/src/main/java/io/quarkus/devservices/mysql/deployment/MySQLDevServicesProcessor.java
+++ b/extensions/devservices/mysql/src/main/java/io/quarkus/devservices/mysql/deployment/MySQLDevServicesProcessor.java
@@ -32,17 +32,17 @@ public class MySQLDevServicesProcessor {
             public RunningDevServicesDatasource startDatabase(Optional<String> username, Optional<String> password,
                     Optional<String> datasourceName, Optional<String> imageName, Map<String, String> additionalProperties,
                     OptionalInt fixedExposedPort, LaunchMode launchMode) {
-                MySQLContainer container = new QuarkusMySQLContainer(imageName, fixedExposedPort,
-                        devServicesSharedNetworkBuildItem.isPresent())
-                                .withPassword(password.orElse("quarkus"))
-                                .withUsername(username.orElse("quarkus"))
-                                .withDatabaseName(datasourceName.orElse("default"));
+                QuarkusMySQLContainer container = new QuarkusMySQLContainer(imageName, fixedExposedPort,
+                        devServicesSharedNetworkBuildItem.isPresent());
+                container.withPassword(password.orElse("quarkus"))
+                        .withUsername(username.orElse("quarkus"))
+                        .withDatabaseName(datasourceName.orElse("default"));
                 additionalProperties.forEach(container::withUrlParam);
                 container.start();
 
                 LOG.info("Dev Services for MySQL started.");
 
-                return new RunningDevServicesDatasource(container.getJdbcUrl(), container.getUsername(),
+                return new RunningDevServicesDatasource(container.getEffectiveJdbcUrl(), container.getUsername(),
                         container.getPassword(),
                         new Closeable() {
                             @Override
@@ -83,8 +83,9 @@ public class MySQLDevServicesProcessor {
             }
         }
 
-        @Override
-        public String getJdbcUrl() {
+        // this is meant to be called by Quarkus code and is needed in order to not disrupt testcontainers
+        // from being able to determine the status of the container (which it does by trying to acquire a connection)
+        public String getEffectiveJdbcUrl() {
             if (useSharedNetwork) {
                 // in this case we expose the URL using the network alias we created in 'configure'
                 // and the container port since the application communicating with this container

--- a/extensions/devservices/postgresql/src/main/java/io/quarkus/devservices/postgresql/deployment/PostgresqlDevServicesProcessor.java
+++ b/extensions/devservices/postgresql/src/main/java/io/quarkus/devservices/postgresql/deployment/PostgresqlDevServicesProcessor.java
@@ -32,18 +32,18 @@ public class PostgresqlDevServicesProcessor {
             public RunningDevServicesDatasource startDatabase(Optional<String> username, Optional<String> password,
                     Optional<String> datasourceName, Optional<String> imageName, Map<String, String> additionalProperties,
                     OptionalInt fixedExposedPort, LaunchMode launchMode) {
-                PostgreSQLContainer container = new QuarkusPostgreSQLContainer(imageName, fixedExposedPort,
-                        devServicesSharedNetworkBuildItem.isPresent())
-                                .withPassword(password.orElse("quarkus"))
-                                .withUsername(username.orElse("quarkus"))
-                                .withDatabaseName(datasourceName.orElse("default"));
+                QuarkusPostgreSQLContainer container = new QuarkusPostgreSQLContainer(imageName, fixedExposedPort,
+                        devServicesSharedNetworkBuildItem.isPresent());
+                container.withPassword(password.orElse("quarkus"))
+                        .withUsername(username.orElse("quarkus"))
+                        .withDatabaseName(datasourceName.orElse("default"));
                 additionalProperties.forEach(container::withUrlParam);
 
                 container.start();
 
                 LOG.info("Dev Services for PostgreSQL started.");
 
-                return new RunningDevServicesDatasource(container.getJdbcUrl(), container.getUsername(),
+                return new RunningDevServicesDatasource(container.getEffectiveJdbcUrl(), container.getUsername(),
                         container.getPassword(),
                         new Closeable() {
                             @Override
@@ -84,8 +84,11 @@ public class PostgresqlDevServicesProcessor {
             }
         }
 
-        @Override
-        public String getJdbcUrl() {
+        // this is meant to be called by Quarkus code and is not strictly needed
+        // in the PostgreSQL case as testcontainers does not try to establish
+        // a connection to determine if the container is ready, but we do it anyway to be consistent across
+        // DB containers
+        public String getEffectiveJdbcUrl() {
             if (useSharedNetwork) {
                 // in this case we expose the URL using the network alias we created in 'configure'
                 // and the container port since the application communicating with this container

--- a/integration-tests/container-image/maven-invoker-way/pom.xml
+++ b/integration-tests/container-image/maven-invoker-way/pom.xml
@@ -138,6 +138,9 @@
                         </goals>
                     </execution>
                 </executions>
+                <configuration>
+                    <skip>${skipTests}</skip>
+                </configuration>
             </plugin>
         </plugins>
     </build>

